src/commands/migrate/list.mts

```diff
@@ -33,6 +38,12 @@ const BaseListCommand = {
 		for (const migration of migrations) {
 			consola.log(`[${migration.executedAt ? '`✓`' : ' '}] ${migration.name}`)
 		}
+
+		const hasPending = migrations.some(migration => migration.executedAt === undefined);
+
+		if (context.args['fail-on-pending'] && hasPending) {
+			throw new Error('Failed due to pending migrations.');
+		}
```

Comment:
I'd exit early when it's not relevant:

```ts
if (!context.args['fail-on-pending']) {
  return;
}
```

and use the existing `exitWithError` helper: 

```ts
if (hasPending) {
  exitWithError(new Error('Failed due to pending migrations.'))
}
```

Reply:
@igalklebanov done! when i use `exitWithError` it does still exit with status code 1, but no longer prints the reason why the command filed to the CLI, is that expected?